### PR TITLE
Event owner selector refactored a bit

### PR DESCRIPTION
**Pull request by antban**
ARUHA-2663 - make event owner selector more clean

### CODE REVIEW

File: core-common/src/main/java/org/zalando/nakadi/domain/EventAuthField.java [hunk 1]
```diff
@@ -11,26 +11,29 @@
...
-        record.headers().add(HEADER_KEY_CLASSIFIER, classifier.getBytes(Charsets.UTF_8));
+        record.headers().add(AUTH_PARAM_NAME, name.getBytes(Charsets.UTF_8));
+        record.headers().add(AUTH_PARAM_VALUE, value.getBytes(Charsets.UTF_8));
     }
 
     public static EventAuthField deserialize(final ConsumerRecord<byte[], byte[]> record) {
-        final Header typeHeader = record.headers().lastHeader(EventAuthField.HEADER_KEY_TYPE);
-        final Header valueHeader = record.headers().lastHeader(EventAuthField.HEADER_KEY_CLASSIFIER);
-        if (typeHeader == null || valueHeader == null) {
+        final Header typeHeader = record.headers().lastHeader(EventAuthField.AUTH_PARAM_NAME);
```
Comment (adyach) [inline core-common/src/main/java/org/zalando/nakadi/domain/EventAuthField.java]: s/type/name

File: core-common/src/test/java/org/zalando/nakadi/domain/EventAuthFieldTest.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-public class EventAuthHeaderTest {
+public class EventAuthFieldTest {
 
     private static final String TYPE = "teams";
     public static final String CLASSIFIER = "nakadi";
```
Comment (adyach) [inline core-common/src/test/java/org/zalando/nakadi/domain/EventAuthFieldTest.java]: value

File: docs/_documentation/using_per_event_authorization.md [hunk 3]
```diff
@@ -3,27 +3,30 @@ title: Event-Based Authorization
...
- by an authorization plugin to classify if a consumer should receive an event.
- If the field is absent or set to `null` in a published event, it is available for all consumers to read.
-
- An `event_auth_field` field can be used to specify a classifier (which along with the authorization section)
- defines who can read a published event.
- An `event_auth_field` in an event type definition can look, for instance like:
+ The `event_owner_selector` defines following values: 
+  - `type` - the way how nakadi will extract owner from published events
+  - `name` - the name of authorization_parameter that will be extracted and stored with event. 
+  This name is used ad `AuthorizationAttribute` data_type for security checks with authz plugin.
```
Comment (adyach) [inline docs/_documentation/using_per_event_authorization.md]: `ad` ?

File: docs/_documentation/using_per_event_authorization.md [hunk 4]
```diff
@@ -48,3 +51,18 @@ An event may be published to the above event type, and the logic for authorizati
 ```
 
  Also, once a `event_auth_field` is specified for an event type, it cannot be removed or updated.
+
+ There are following event owner selector types supported: 
+  - `path` - dot separated path within published event (after enrichment), in this case `value` 
+  should hold dot separated path to a field that will be used as `AuthorizationParameter` value
```
Comment (adyach) [inline docs/_documentation/using_per_event_authorization.md]: `.` at the end

File: docs/_documentation/using_per_event_authorization.md [hunk 5]
```diff
@@ -48,3 +51,18 @@ An event may be published to the above event type, and the logic for authorizati
...
 
  Also, once a `event_auth_field` is specified for an event type, it cannot be removed or updated.
+
+ There are following event owner selector types supported: 
+  - `path` - dot separated path within published event (after enrichment), in this case `value` 
+  should hold dot separated path to a field that will be used as `AuthorizationParameter` value 
+  - `static` - all events, that are published to nakadi will have the same `AuthorizationParameter` 
+  value, equal to `event_owner_selector` `value`   field.
+  
+ During consumption, the consumer is checked through authz plugin whether or not it is allowed to
```
Comment (adyach) [inline docs/_documentation/using_per_event_authorization.md]: s/authz/authorization

**Comment by antban** [comment]
@adyach fixed: https://github.com/zalando/nakadi/pull/1145/commits/6e0a041e13de1a780cc509bfbab6f86d3e90d40b

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

**Pull request merged**